simtools/utils/general.py

```diff
@@ -328,6 +332,63 @@ def _validate_and_convert_value(par_name, par_info, value_in):
     return _validate_and_convert_value_with_units(value, value_keys, par_name, par_info)
 
 
+def is_url(url):
+    """
+    Check if a string is a valid URL.
+
+    Parameters
+    ----------
+    url: str
+        String to be checked.
+
+    Returns
+    -------
+    bool
+        True if url is a valid URL.
+
+    """
+
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc])
+    except ValueError:
+        return False
+
+
+def collect_data_from_http_yaml(url):
+    """
+    Download yaml file from url and return it contents as dict.
+    File is downloaded as a temporary file and deleted afterwards.
+
+    Parameters
+    ----------
+    url: str
+        URL of the yaml file.
+
+    Returns
+    -------
+    dict
+        Dictionary containing the yaml file contents.
+
+    """
+
+    _logger.debug(f"Downloaded yaml file from {url}")
+    try:
+        with tempfile.NamedTemporaryFile() as tmp_file:
+            urllib.request.urlretrieve(url, tmp_file.name)
+            data = yaml.load(tmp_file)
```

Comment:
should we use our general function ` collect_data_from_yaml_or_dict` to open this, instead of calling yaml.load?

Reply:
I am hesitating, as this function(`collect_data_from_http_yaml`) is actually called from `collect_data_from_yaml_or_dict`.